sdcm/utils/common.py

```diff
@@ -98,12 +108,24 @@ def inner(*args, **kwargs):
                 except self.allowed_exceptions as ex:
                     LOGGER.debug("'%s': failed with '%r', retrying [#%s]", func.__name__, ex, i)
                     time.sleep(self.sleep_time)
-                    if i == self.n - 1:
-                        LOGGER.error("'%s': Number of retries exceeded!", func.__name__)
+                    if i == self.n - 1 or (end_time and time.time() > end_time):
+                        LOGGER.error(f"'{func.__name__}': Number of retries exceeded!")
```

Comment:
need to distinguish between timeout and number of retries so it will be clear to the user